### ISSUE TITLE
Redirects do not restore query arguments if they had to be removed to find a valid redirect

### ISSUE THREAD

**Issue by Pomax** [wagtail/wagtail#7339]
Found a bug? Please fill out the sections below. 👍

### Issue Summary

In https://github.com/wagtail/wagtail/blob/main/wagtail/contrib/redirects/middleware.py#L43-L55, the code removes query arguments to find a redirect:

```python
        # Find redirect
        redirect = get_redirect(request, path)
        if redirect is None:
            # Get the path without the query string or params
            path_without_query = urlparse(path).path

            if path == path_without_query:
                # don't try again if we know we will get the same response
                return response

            redirect = get_redirect(request, path_without_query)
            if redirect is None:
                return response
```
However, we also see that if it finds redirect after removing query arguments, it does not then restore the query arguments for the redirect target. Those arguments have now been lost.

**Comment by laymonage** [comment]
Thank you for the report! This looks like a duplicate of #4414, so closing.

**Comment by Pomax** [comment]
Makes sense, but then at least please also fix it if it's still an issue rather than letting an even older issue with a seven year old PR just disappear into the void. This should be a pretty quick patch to write and test for anyone who works with python on a daily basis =P
